sky/templates/gcp-ray.yml.j2

```diff
@@ -64,6 +64,9 @@ provider:
   # leakage.
   disable_launch_config_check: true
   use_managed_instance_group: {{ gcp_use_managed_instance_group }}
+{%- if enable_gvnic %}
+  enable_gvnic: {{ enable_gvnic }}
+{%- endif %}
```

Comment:
nit: should this be in `provider_config` or in the `node_config` below. I would vote for adding to the `node_config` below instead.

Reply:
I wanted to add to `node_config` too, but realized `node_config` mainly contains fields that are directly passed to gcp APIs (bulk_insert). Adding to `node_config` would require special casing with a .pop('enable_gvnic') in downstream methods. Decided it's best to avoid non-GCP API fields in `node_config`.